cpp/mrc/src/tests/common.hpp

```diff
@@ -17,10 +17,30 @@
 
 #pragma once
 
+#include <gtest/gtest.h>
+
+#include <cstdlib>
 #include <functional>
 #include <memory>
 
 namespace mrc {
+
+#define SKIP_IF_IN_CI()                           \
+    {                                             \
+        if (std::getenv("CI") != nullptr)         \
+        {                                         \
+            GTEST_SKIP() << "Test skipped in CI"; \
+        }                                         \
+    }
+
+#define SKIP_IF_CODE_COV()                          \
+    {                                               \
+        if (std::getenv("CODE_COV_RUN") != nullptr) \
+        {                                           \
+            GTEST_SKIP() << "Test skipped in CI";   \
+        }                                           \
+    }
+
```

Comment:
Since this variable is only set in the CI scripts, this will give different coverage results when running locally vs in CI. Instead, since we enable code coverage with the `MRC_ENABLE_CODECOV` CMake option, we should use a `#define` directive set by CMake. You would need an option set in the libmrc CMakeLists.txt:

```cmake
if (MRC_ENABLE_CODECOV)
  target_compile_definitions(libmrc INTERFACE "MRC_CODECOV_ENABLED")
endif()
```

And then the macro becomes:
```cpp
#ifdef MRC_CODECOV_ENABLE
    #define SKIP_IF_CODE_COV() GTEST_SKIP() << "Skipping test when code coverage is enabled";
#else
    #define SKIP_IF_CODE_COV()
#endif
```

Reply:
good call